lib/ChatBot.jsx

```diff
@@ -294,7 +339,7 @@ class ChatBot extends Component {
       delete currentStep.options;
 
       // Find the last state and append it to the new one
-      const lastSameSteps = previousSteps.filter(step => step.id === currentStep.id);
+      const lastSameSteps = renderedSteps.filter(step => step.id === currentStep.id);
```

Comment:
What's the difference between `previousSteps` and `renderedSteps`?

Reply:
None. as far as the contents inside them is concerned. But, during rendering and "extracting" data, `renderedSteps` is used instead of `previousSteps`. I lost some considerable amount of time debugging the code thinking appending only to `previousSteps` would do the trick, when appending only to `renderedSteps` would have done the trick. Nevertheless, current implementation appends to both.

As a note, in future, any body working on this should always try to change both variables.

As for why there is two different variables that contain the same data, I don't know. I checked every mention of `previousSteps` and `renderedSteps` but the only difference is that `renderedSteps` is used for rendering while `previousSteps` is not used. Thus, it is my opinion that we could eliminate the use of `previousSteps` altogether. Perhaps, I could be wrong.